pkg/cephfs/controllerserver.go

```diff
@@ -184,7 +184,7 @@ func (cs *ControllerServer) deleteVolumeDeprecated(req *csi.DeleteVolumeRequest)
 	mtxControllerVolumeID.LockKey(string(volID))
 	defer mustUnlock(mtxControllerVolumeID, string(volID))
 
-	if err = purgeVolume(volID, cr, &ce.VolOptions); err != nil {
+	if err = purgeVolume(volID, &ce.VolOptions); err != nil {
```

Comment:
> * Where will the `subvolume rm` command look for the passed in <CSI volID/name>? @ajarr
>   "Where" as in which path, I would assume based on [this](https://github.com/ceph/ceph-csi/pull/400#issuecomment-500358206) comment this would be, `/volumes/_nogroup/<csi-vol-id>` (and possibly in the future `/volume/csi/<csi-vol-id>`)

Correct.

> * 1.0 created volumes would not be in the said location, unless there is more smartness built into the `ceph fs subvolume ...` commands implementation (which I do not expect). @ajarr
> 

No, the `ceph fs subvolume` doesn't remove the earlier version of subvolume that had a different path.

Reply:
> > * 1.0 created volumes would not be in the said location, unless there is more smartness built into the `ceph fs subvolume ...` commands implementation (which I do not expect). @ajarr
> 
> No, the `ceph fs subvolume` doesn't remove the earlier version of subvolume that had a different path.

@poornimag we would need to invoke the older purge routines to delete and purge volumes created using v1.0 and earlier. Current code would break the backward compatibility otherwise.

Reply:
@Madhu-1 @humblec Suggested to not have backward compatibility in the code. IMO, It will be easier to have backward compatibility than migrating old volumes to new paths, as i m not sure what happens to the attributes set, user permission also needs to be modified in that case. Thoughts?
Also Is it alright if its a different PR?

Reply:
@poornimag considering this is a breaking change which is inevitable, the backward compatibility can be taken care later.  We could think about a documentation or different PR if at all we have a  mechanism for backward compatibility.

Reply:
As long as we ensure we release v1.1.0 with the backward compatibility I am fine it being a different PR (this is the same I did with stateless+RBD and the compat PR is being tracked for v1.1.0 #391 )